spec/datetimeformat.html

```diff
@@ -440,6 +462,126 @@ <h1>FormatDateTimeToParts ( _dateTimeFormat_, _x_ )</h1>
           1. Increment _n_ by 1.
         1. Return _result_.
       </emu-alg>
+
+    </emu-clause>
+
+    <emu-clause id="sec-partitiondatetimerangepattern" aoid="PartitionDateTimeRangePattern">
+      <h1>PartitionDateTimeRangePattern ( _dateTimeFormat_, _x_, _y_ )</h1>
+
+      <p>
+        The PartitionDateTimeRangePattern abstract operation is called with arguments _dateTimeFormat_ (which must be an object initialized as a DateTimeFormat), _x_ (which must be a Number value) and _y_ (which must be a Number value), interprets _x_ and _y_ as time values as specified in ES2021, <emu-xref href="#sec-time-values-and-time-range"></emu-xref>, and creates the corresponding parts according to the effective locale and the formatting options of _dateTimeFormat_. The following steps are taken:
+      </p>
+      <emu-alg>
+        1. Let _x_ be TimeClip(_x_).
+        1. If _x_ is *NaN*, throw a *RangeError* exception.
+        1. Let _y_ be TimeClip(_y_).
+        1. If _y_ is *NaN*, throw a *RangeError* exception.
+        1. If _x_ is greater than _y_, throw a *RangeError* exception.
+        1. Let _tm1_ be ToLocalTime(_x_, _dateTimeFormat_.[[Calendar]], _dateTimeFormat_.[[TimeZone]]).
+        1. Let _tm2_ be ToLocalTime(_y_, _dateTimeFormat_.[[Calendar]], _dateTimeFormat_.[[TimeZone]]).
+        1. Let _rangePatterns_ be _dateTimeFormat_.[[RangePatterns]].
+        1. Let _rangePattern_ be *undefined*.
+        1. Let _dateFieldsPracticallyEqual_ be *true*.
+        1. Let _patternContainsLargerDateField_ be *false*.
+        1. While _dateFieldsPracticallyEqual_ is *true* and _patternContainsLargerDateField_ is *false*, repeat for each row of <emu-xref href="#table-datetimeformat-rangepatternfields"></emu-xref> in order, except the header row
```

Comment:
`expected freeform line with substeps to end with ":" (found " in order, except the header row")`